contracts/InterchainTokenService.sol

```diff
@@ -623,6 +624,29 @@ contract InterchainTokenService is
         }
     }
 
+    /**
+     * @notice Allows the owner to migrate legacy tokens that cannot be migrated automatically.
+     * @param tokenId the tokenId of the registered token.
+     * @param data the data to pass to migrate the token.
+     */
+    function migrateLegacyToken(bytes32 tokenId, bytes calldata data) external onlyOwner {
+        address tokenAddress = registeredTokenAddress(tokenId);
+        (bool success, bytes memory returnData) = tokenAddress.call(data);
+        if (!success) {
+            revert TokenMigrateFailed(returnData);
+        }
+    }
+
+    /**
+     * @notice Allows the owner to migrate legacy tokens that cannot be migrated automatically.
+     * @param tokenId the tokenId of the registered token.
+     */
+    function migrateInterchainToken(bytes32 tokenId) external onlyOwner {
+        ITokenManager tokenManager_ = deployedTokenManager(tokenId);
+        address tokenAddress = tokenManager_.tokenAddress();
+        IMinter(tokenAddress).transferMintership(address(tokenManager_));
```

Comment:
for unit test, also test failure when migrating an already migrated token

Reply:
added all tests, please check